code/game/supplyshuttle.dm

```diff
@@ -688,6 +874,272 @@ var/datum/controller/supply/supply_controller = new()
 		</p></div>"
 
 	name = "[name] - [ordername]"
+
+/datum/supply_ui
```

Comment:
I imagine that the next steps here are to refactor this work into some sort of control computer. It should probably be in its own file